examples/hybrid/sphere/deformation_flow.jl

```diff
@@ -23,412 +25,306 @@ import TerminalLoggers
+        q5 = FT(1)
+
+        ρ = ρ_ref(z)
+        (; ρ, ρq = (ρ * q1, ρ * q2, ρ * q3, ρ * q4, ρ * q5))
+    end
+    Y = Fields.FieldVector(; c = cent_Y)
+
+    cache = (;
+        cent_uvw = Fields.Field(Geometry.Covariant123Vector{FT}, cent_space),
+        face_uvw = Fields.Field(Geometry.Covariant123Vector{FT}, face_space),
+        face_uv = Fields.Field(Geometry.Covariant12Vector{FT}, face_space),
+        face_w = Fields.Field(Geometry.Covariant3Vector{FT}, face_space),
+        Δₕq = similar(Y.c.ρq),
+        limiter = use_limiter ? Limiters.QuasiMonotoneLimiter(Y.c.ρq) : nothing,
+        fct_op,
+        ρq_td = isnothing(fct_op) ? nothing : similar(Y.c.ρq.:1),
+    )
 
-    Spaces.weighted_dss!(dydt.ρ)
-    Spaces.weighted_dss!(dydt.ρq1)
-    Spaces.weighted_dss!(dydt.ρq2)
-    Spaces.weighted_dss!(dydt.ρq3)
-    Spaces.weighted_dss!(dydt.ρq4)
+    # TODO: Fix ClimaTimeSteppers to use a ForwardEulerFunction here
+    problem =
+        ODEProblem(IncrementingODEFunction(increment!), Y, (0, t_end), cache)
+    return solve(problem, ode_algorithm; dt, saveat = t_end / 2)
+end
 
-    return dydt
+function conservation_errors(sol)
+    initial_total_mass = sum(sol.u[1].c.ρ)
+    initial_tracer_masses = map(n -> sum(sol.u[1].c.ρq.:($n)), 1:5)
+    final_total_mass = sum(sol.u[end].c.ρ)
+    final_tracer_masses = map(n -> sum(sol.u[end].c.ρq.:($n)), 1:5)
+    return (
+        (final_total_mass - initial_total_mass) / initial_total_mass,
+        (final_tracer_masses .- initial_tracer_masses) ./ initial_tracer_masses,
+    )
 end
 
-# Set up RHS function
-ystar = copy(y0)
-y_td = copy(y0)
+ref_sol = run_deformation_flow(false, nothing)
+lim_sol = run_deformation_flow(true, nothing)
+fct_sol = run_deformation_flow(false, FCTZalesak)
 
-parameters = (; τ, coords, face_coords, ystar, y_td)
+ref_ρ_err, ref_ρq_errs = conservation_errors(ref_sol)
+lim_ρ_err, lim_ρq_errs = conservation_errors(lim_sol)
+fct_ρ_err, fct_ρq_errs = conservation_errors(fct_sol)
 
-rhs!(ystar, y0, parameters, 0.0, dt, 1)
+# Check that the conservation errors are not too big.
+@test abs(ref_ρ_err) < (FT == Float64 ? 1e-14 : 2e-5)
+@test all(abs.(ref_ρq_errs) .< 3 * abs(ref_ρ_err))
 
-# run!
-prob = ODEProblem(IncrementingODEFunction(rhs!), copy(y0), (0.0, T), parameters)
-sol = solve(
-    prob,
-    SSPRK33ShuOsher(),
-    dt = dt,
-    saveat = dt,
-    progress = true,
-    adaptive = false,
-    progress_message = (dt, u, p, t) -> t,
-)
+# Check that the limiter and FCT have no effect on ρ.
+@test ref_ρ_err == lim_ρ_err == fct_ρ_err
+
+# Check that the error of the tracer with q = 1 is nearly the same as that of ρ.
+@test ref_ρq_errs[5] == ref_ρ_err
+@test lim_ρq_errs[5] ≈ lim_ρ_err rtol = 0.02
+@test fct_ρq_errs[5] ≈ fct_ρ_err rtol = 0.02
+
+# Check that the limiter and FCT do not significantly change the tracer errors.
+@test all(.≈(lim_ρq_errs, ref_ρq_errs, rtol = 0.3))
+@test all(.≈(fct_ρq_errs, ref_ρq_errs, rtol = 0.3))
```

Comment:
I think it will be also good to check the `min` and `max` to see the effect of the limiter and FCT

Reply:
The neither the limiter nor FCT consistently decreases/increases the conservation error. All this is checking is that they do not change the conservation error in either direction by more than 30%. I'm not sure what test I would write with min/max.

Reply:
I was thinking if just checking that the min and max of the tracer concentrations per unit mass (`q`) were reduced when applying these (hence, it would be a sign that under/over-shoots diminish), in addition to a visual inspection of the produced artifacts :eyes:

Reply:
Would it be better to check that the limiters improve the "smoothness" of the tracer concentrations? That is, check that `sum(norm.(hgrad.(ρq)))` is decreased? After all, the over-/under-shoots are relative to local bounds on q, rather than relative to the global minimum/maximum.

Reply:
I ended up checking the tracers' arithmetic roughness and their ranges, both of which are reduced by limiters

Reply:
Sounds good. Thank you! I would also check the same for the FCT solution. No?

Reply:
Sure, I can do that. The upper bounds need to be increased to 0.96 and 0.9 for FCT, though.